sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -258,6 +266,10 @@ createTableHeader
     : CREATE TEMPORARY? EXTERNAL? TABLE (IF NOT EXISTS)? multipartIdentifier
     ;
 
+replaceTableHeader
+    : REPLACE TEMPORARY? TABLE multipartIdentifier
```

Comment:
I'd probably remove `TEMPORARY` to begin with. What is the behavior of a temporary table? I think it used to be a view.

Reply:
Actually, it looks fine since this is not allowed in the AST builder.

Reply:
Let's get rid of `TEMPORARY TABLE`. It was a mistake and we've almost removed everything about `TEMPORARY TABLE` in Spark, only a few parser rules are left for backward compatibility reason.

To clarify, there is no `TEMPORARY TABLE` in Spark, it never had. Spark only has TABLE, VIEW and TEMP VIEW.